src/lib/crypto/krb/string_to_key.c

```diff
@@ -51,8 +51,11 @@ krb5_c_string_to_key_with_params(krb5_context context, krb5_enctype enctype,
         return KRB5_BAD_ENCTYPE;
     keylength = ktp->enc->keylength;
 
+    if (salt == NULL)
```

Comment:
I opened PR #622 to address this particular check as discussed above, so we can just eliminate this hunk from this commit.